includes/admin/class-wp-job-manager-admin-notices.php

```diff
@@ -348,31 +395,43 @@ private static function save_dismissed_notices( $dismissed_notices, $is_user_not
 	public static function handle_notice_dismiss() {
 		check_ajax_referer( self::DISMISS_NOTICE_ACTION, 'nonce' );
 
-		$notices   = self::get_notices();
 		$notice_id = isset( $_POST['notice'] ) ? sanitize_text_field( wp_unslash( $_POST['notice'] ) ) : false;
-		if ( ! $notice_id || ! isset( $notices[ $notice_id ] ) ) {
+		if ( ! $notice_id ) {
 			return;
 		}
 
-		$notice = self::normalize_notice( $notices[ $notice_id ] );
+		$notice = self::get_notice( $notice_id );
 
-		$is_dismissible       = $notice['dismissible'];
-		$is_user_notification = 'user' === $notice['type'];
-		if (
-			! $is_dismissible
-			|| ( ! $is_user_notification && ! current_user_can( 'manage_options' ) )
-		) {
+		if ( ! $notice || ! $notice['dismissible'] || ( 'user' !== $notice['type'] && ! current_user_can( 'manage_options' ) ) ) {
 			wp_die( '', '', 403 );
 		}
 
+		self::dismiss_notice( $notice_id );
+		exit;
+	}
+
+	/**
+	 * Save notice state as dismissed.
+	 *
+	 * @param int $notice_id Notice ID.
+	 *
+	 * @return void
+	 */
+	public static function dismiss_notice( $notice_id ) {
```

Comment:
```suggestion
	private static function dismiss_notice( $notice_id ) {
```

Reply:
Left this one to have a public method for devs to dismiss a single notice. It's used now in the install hook with the latest changes so gotta keep it public.